packages/contracts/contracts/CommitmentChain.sol

```diff
@@ -5,39 +5,155 @@ pragma experimental ABIEncoderV2;
 import {DataTypes as dt} from "./DataTypes.sol";
 
 contract CommitmentChain {
+
+    address public aggregator;
+    uint128 public nextPlasmaBlockNumber = 0;
+    mapping (uint128 => bytes32) blocks;
+
+    constructor(address _aggregator) public {
+        aggregator = _aggregator;
+    }
+
+    function getBlockRoot(uint128 _plasmaBlockNumber) public view returns(bytes32) {
+        return blocks[_plasmaBlockNumber];
+    }
+
+    function submitBlock(bytes32 _blockHeader) public {
+        // make sure it's the aggregator submitting the block
+        require(msg.sender == aggregator, 'Only the aggregator can submit blocks.');
+        // store the block
+        blocks[nextPlasmaBlockNumber] = _blockHeader;
+        // increment the block number
+        nextPlasmaBlockNumber += 1;
+    }
+
     function verifyInclusion(dt.StateUpdate memory _stateUpdate, bytes memory _inclusionProof) public returns (bool) {
         // Always return true for now until we can verify inclusion proofs.
         return true;
     }
 
-    function getAbiPacked(
-        dt.StateSubtreeNode memory _leftSibling,
-        dt.StateSubtreeNode memory _rightSibling
-    ) public pure returns (bytes memory) {
-        bytes memory packed =
+    function verifyStateUpdateInclusion(dt.StateUpdate memory _stateUpdate, dt.StateUpdateInclusionProof memory _inclusionProof) public view returns (bool) {
+        // calculate what the state root for the subtree should be from the stateTreeInclusionProof
+        bytes32 stateSubtreeRoot = verifySubtreeInclusionAndGetRoot(_stateUpdate, _inclusionProof.stateTreeInclusionProof);
+        // calculate what the block root would be if the stateSubtreeRoot is correct
+        bytes32 assetTreeRoot = verifyAssetTreeInclusionAndGetRoot(stateSubtreeRoot, _stateUpdate.depositAddress, _inclusionProof.assetTreeInclusionProof);
+        // check this is indeed the root for the state update's block
+        uint128 expectedPlasmaBlockNumber = _stateUpdate.plasmaBlockNumber;
+        return assetTreeRoot == blocks[expectedPlasmaBlockNumber];
+    }
+
+    // Via https://github.com/ethereum/solidity-examples/blob/master/src/bits/Bits.sol
+    // Gets the value of the 'index'th bit (where 0 => rightmost bit) in the binary expression of 'self'.
+    function getNthBitFromRight(uint self, uint8 index) public pure returns (uint8) {
```

Comment:
Looks like the result of this is used as a `bool` everywhere -- should the name change to `isNthBitFromRightSet` and the body change to `return (self >> index & 1) == 1`?